sql-plugin/src/main/311+-all/scala/com/nvidia/spark/rapids/shims/v2/GpuRowBasedScalaUDF.scala

```diff
@@ -55,10 +55,15 @@ case class GpuRowBasedScalaUDF(
    *         created by using `ExpressionEncoder`.
    */
   private def scalaConverter(i: Int, dataType: DataType): (Any => Any, Boolean) = {
+    val arrayWithNull = dataType match {
+      case ArrayType(_, containsNull) => containsNull
+      case _ => false
+    }
     val useEncoder =
       !(inputEncoders.isEmpty || // for untyped Scala UDF and Java UDF
         inputEncoders(i).isEmpty || // for types aren't supported by encoder, e.g. Any
-        inputPrimitives(i)) // for primitive types
+        inputPrimitives(i) || // for primitive types
+        arrayWithNull) // https://github.com/NVIDIA/spark-rapids/issues/3942#issuecomment-957146443
```

Comment:
I thought the Catalyst converter doesn't handle some UDFs that would expect a structured class as input and therefore the Catalyst converters are not appropriate?  This change seems like it could break some UDFs since an input encoder exists and we're not using it when Apache Spark would.

Reply:
Yes, you are right.
This change could break some UDFs with the array element types not supported by Catalyst converters.

So, seems we need to wait for the Spark fix.  I will close this PR. Thanks.